### PR TITLE
Fix: Enhance MergingIterator bidirectional traversal and robustness

### PR DESCRIPTION
**Pull request by shanenoi**
This pull request addresses several issues and enhances the functionality of the `MergingIterator` and related components, focusing on improving bidirectional traversal and overall robustness.

**Key Changes:**

*   **`MergingIterator` Bidirectional Traversal:**
    *   Introduced `forward` and `crossingAnchor` fields to `MergingIterator` to correctly handle oscillations between `Next()` and `Prev()` calls. This prevents duplicate record surfacing and ensures accurate iteration when switching directions.
    *   Refined `prepareNext()` and `preparePrev()` to correctly invalidate the opposite direction's prepared state and manage the `crossingAnchor`.
    *   Implemented `matchesCrossingAnchor` helper to accurately detect when an item being processed matches the anchoring point for direction changes.
    *   Added comprehensive tests for `MergingIterator` to cover complex next/prev sequences and edge cases.

*   **`RangeIterator` Improvements:**
    *   Ensured `RangeIterator` correctly leverages the enhanced bidirectional capabilities of `MergingIterator`.
    *   Added tests to verify `RangeIterator`'s behavior with alternating `Next()` and `Prev()` calls, and its handling of `Prev()` before `Next()`.

*   **Test Case Renaming:**
    *   Refactored numerous test function names across various packages (`diffharness`, `merging_iterator`, `range`, `rindb`, `sstable_iteration`, `sstable`, `sstablemgmt`, `tablecache`) to be more descriptive and follow a consistent `[Component]_[Functionality]` or `[Component]_[Scenario]` pattern (e.g., `TestPutOpLoggingAndError` renamed to `TestPutOp_LoggingAndError`). This improves test readability and maintainability.

*   **Other Minor Fixes and Refinements:**
    *   Addressed minor edge cases and error handling in various components.

**Impact:**

This PR significantly improves the reliability and correctness of bidirectional iteration, particularly when switching between forward and backward traversal. The enhanced robustness of `MergingIterator` and `RangeIterator` will benefit any part of the system that relies on these components for data exploration. The test renaming contributes to better code organization and understanding.

**Testing Done:**

*   All existing tests have been run and passed.
*   New tests have been added to cover the bidirectional traversal logic in `MergingIterator` and `RangeIterator`.

### CODE REVIEW

File: merging_iterator.go [hunk 1]
```diff
@@ -229,3 +280,25 @@ func (m *MergingIterator) Close() error {
...
+		return false
+	}
+	if item.rec.GetType() != m.crossingAnchor.rec.GetType() {
+		return false
+	}
+	if item.rec.GetKey().Compare(m.crossingAnchor.rec.GetKey()) != CmpEqual {
+		return false
+	}
+	return true
+}
```
[the remark this answers is not in the document]
Comment (shanenoi) [reply merging_iterator.go:304]: @codex please resolve above comment, i see that there are 2 files has same matchesCrossingAnchor concept:
merging_iterator.go and range.go
